bloom-filter/src/main/scala/bloomfilter/mutable/BloomFilter.scala

```diff
@@ -12,17 +12,27 @@ class BloomFilter[T] private (val numberOfBits: Long, val numberOfHashes: Int, p
     this(numberOfBits, numberOfHashes, new UnsafeBitArray(numberOfBits))
   }
 
+ /** Add an item in the Bloom filter
+   *
+   * @param x the T item to add
+   * @return true if the item was not in the filter before, false otherwise
+   */
   def add(x: T): Unit = {
     val hash = canGenerateHash.generateHash(x)
     val hash1 = hash >>> 32
     val hash2 = (hash << 32) >> 32
+    var was_defined = true
 
     var i = 0
     while (i < numberOfHashes) {
       val computedHash = hash1 + i * hash2
+      if (!bits.get((computedHash & Long.MaxValue) % numberOfBits))
```

Comment:
Why not for the separate method, `addAndGet` is not that bad (that's my way of saying that I can't think of anything clearer).
Do you think it's better to go the naive way
```scala
def addAndGet(x: T): Boolean = {
  val was_set = mightContain(x)
  add(x)
  !was_set
}
```
or C&P?